frontend/src/views/SecretMainPage/components/SecretListView/CreateReminderForm.tsx

```diff
@@ -0,0 +1,117 @@
+import { useEffect } from "react";
+import { useForm } from "react-hook-form";
+import { faClock } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { twMerge } from "tailwind-merge";
+import { z } from "zod";
+
+import { Button, FormControl, Input, Modal, ModalContent, TextArea } from "@app/components/v2";
+
+const ReminderFormSchema = z.object({
+  note: z.string().optional(),
+  days: z
+    .number()
+    .min(1, { message: "Must be at least 1 day" })
+    .max(365, { message: "Must be less than 365 days" })
+});
+export type TReminderFormSchema = z.infer<typeof ReminderFormSchema>;
+
+interface ReminderFormProps {
+  isOpen: boolean;
+  onOpenChange: (isOpen: boolean, data?: TReminderFormSchema) => void;
+}
+
+export const CreateReminderForm = ({ isOpen, onOpenChange }: ReminderFormProps) => {
+  const {
+    register,
+    watch,
+    reset,
+    setValue,
+    handleSubmit,
+    formState: { errors, isSubmitting }
+  } = useForm<TReminderFormSchema>({
+    resolver: zodResolver(ReminderFormSchema)
+  });
+
+  const daysWatch = watch("days");
+
+  const handleFormSubmit = async (data: TReminderFormSchema) => {
+    onOpenChange(false, data);
+  };
+
+  useEffect(() => {
+    if (isOpen) {
+      reset();
+    }
+  }, [isOpen]);
+
+  return (
+    <Modal isOpen={isOpen} onOpenChange={onOpenChange}>
+      <ModalContent
+        title="Create secret reminder"
+        // ? QUESTION: Should this specifically say its for secret rotation?
+        // ? Or should we be call it something more generic?
+        subTitle={
+          <div>
+            Set up a reminder for when this secret should be rotated. Everyone with access to this
+            project will be notified when the reminder is triggered.
+          </div>
+        }
+      >
+        <form onSubmit={handleSubmit(handleFormSubmit)}>
+          <div className="space-y-2">
+            <div>
```

Comment:
U can wrap this whole(input and below label) in controller component of react hook form to avoid using setValue and gettt